### ISSUE TITLE
error when use one step install.sh

### ISSUE THREAD

**Issue by xushaoyi** [Metannotate/Metannotate#73]
hi, author, I have some problems when use the metannotate. would you like to offer me some help? thanks very much!!
here is the details: it report that gi_taxid_prot.dmp has been aborted， so could you please update the code? thans very much!
![image](https://user-images.githubusercontent.com/40730975/91512222-1a6a1900-e914-11ea-9a5f-615bce1f87d6.png)

**Comment by jmtsuji** [comment]
Hi @xushaoyi ,

Thanks for your interest in MetAnnotate. It looks like your issue might have to do with the version of RefSeq you are using. Unfortunately, MetAnnotate is not being actively maintained these days (the original authors moved on from the lab a few years ago). Newer versions of the RefSeq database without GI numbers do not work with MetAnnotate.

Can you try navigating to our development branch and installing MetAnnotate from there, using the instructions in the README? https://github.com/MetAnnotate/MetAnnotate/tree/develop In that branch, we froze the version of RefSeq to ensure that it is compatible with MetAnnotate, and we also streamlined the overall install process.

The recommended install method is using Docker, if you have it on your server: https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui In my experience, this pretty much always works.

Otherwise, I recommend only installing the command-line interface: https://github.com/MetAnnotate/MetAnnotate/tree/develop#command-line-non-docker-installation

Hope this helps. Our long-term hope is to do a major code update to MetAnnotate. I might be able to work on this after I finish my PhD in a few weeks.

**Comment by xushaoyi** [comment]
> Hi @xushaoyi ,
> 
> Thanks for your interest in MetAnnotate. It looks like your issue might have to do with the version of RefSeq you are using. Unfortunately, MetAnnotate is not being actively maintained these days (the original authors moved on from the lab a few years ago). Newer versions of the RefSeq database without GI numbers do not work with MetAnnotate.
> 
> Can you try navigating to our development branch and installing MetAnnotate from there, using the instructions in the README? https://github.com/MetAnnotate/MetAnnotate/tree/develop In that branch, we froze the version of RefSeq to ensure that it is compatible with MetAnnotate, and we also streamlined the overall install process.
> 
> The recommended install method is using Docker, if you have it on your server: https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui In my experience, this pretty much always works.
> 
> Otherwise, I recommend only installing the command-line interface: https://github.com/MetAnnotate/MetAnnotate/tree/develop#command-line-non-docker-installation
> 
> Hope this helps. Our long-term hope is to do a major code update to MetAnnotate. I might be able to work on this after I finish my PhD in a few weeks.

Thanks for your reply！ infact I am new to docker, but I tried to install the software by it. here is the details, did I do something wrong? it give an error and I don't know what's happened.
![image](https://user-images.githubusercontent.com/40730975/91711401-69d07380-ebb8-11ea-8465-d52309d7472d.png)

**Comment by jmtsuji** [comment]
Hi @xushaoyi It looks like Docker is not installed on your machine or is not functioning properly. Docker is a requirement for `enter-metannotate` to run.

Installation instructions are here: https://docs.docker.com/get-docker/

Try running `docker run hello-world` to see if Docker is working. It should print a friendly welcome message.

**Comment by xushaoyi** [comment]
> Hi @xushaoyi It looks like Docker is not installed on your machine or is not functioning properly. Docker is a requirement for `enter-metannotate` to run.
> 
> Installation instructions are here: https://docs.docker.com/get-docker/
> 
> Try running `docker run hello-world` to see if Docker is working. It should print a friendly welcome message.

thanks very much! I have downloaded the refeseq db as you suggested, it's helpful, and the example ran successfully, but I have a small question, how can I identify the abudance of RPOB from raw metagenome sequences which was paired-end sequenced? it seems the run command didn't have the option to input the paired-end sequences which has two fasta files, but I read from the literature that described as follows: "Profile hidden Markov models (HMMs) for taxonomic marker (i.e., rpoB) and functional
genes (i.e., amoA_AOA, amoA_AOB, and nxrB), downloaded from FunGene , were used to quantify **the relative abundances and taxonomic affiliations of nitrifiers from the unassembled reads** using MetAnnotate" , so I wonder if there were any suggestion from you, how can I achieve that, hope for your reply!
![image](https://user-images.githubusercontent.com/40730975/92547818-598b5900-f288-11ea-8dc3-486bb09f3a78.png)

**Comment by jmtsuji** [comment]
@xushaoyi I'm glad that MetAnnotate worked for you! Are you following the Spasov et al., 2020 paper?

I recommend just running MetAnnotate on the R1 (forward) reads for metagenomes with paired-end reads -- i.e., just on one FastA file per metagenome. If you run MetAnnotate on both the forward (R1) and reverse (R2) reads, then the statistics get a little complicated, because the R1 and R2 reads are pseudo-replicates of each other linked based on distance on the genome.

If you want to use the TSV output file of MetAnnotate to calculate/visualize the relative abundances of taxa in the hits for your genes of interest, you might try using the metannoviz R library within an R console: https://github.com/MetAnnotate/metannoviz

**Comment by xushaoyi** [comment]
> @xushaoyi I'm glad that MetAnnotate worked for you! Are you following the Spasov et al., 2020 paper?
> 
> I recommend just running MetAnnotate on the R1 (forward) reads for metagenomes with paired-end reads -- i.e., just on one FastA file per metagenome. If you run MetAnnotate on both the forward (R1) and reverse (R2) reads, then the statistics get a little complicated, because the R1 and R2 reads are pseudo-replicates of each other linked based on distance on the genome.
> 
> If you want to use the TSV output file of MetAnnotate to calculate/visualize the relative abundances of taxa in the hits for your genes of interest, you might try using the metannoviz R library within an R console: https://github.com/MetAnnotate/metannoviz

yeah, I  followed the analysis pipeline used by Spasov et al., 2020, oh, I understand, so just needed to use the sequenced output file(fasta format) directly to do the metanotate analysis? also I wonder if I can run several HMM hits simultaneously, is the follows right code? 
"python run_metannotate.py --do_transeq --filter_multi_orf --filter_multi_refseq --orf_files=/home/ming/XSY/MBBR8samples/fasta/A1_1.fasta \
--hmm_files=data/hmms/RPOB.HMM amoA_AOB.HMM nxrB.HMM  --reference_database=data/metannotate_refseq_db_w_gi_2017_03_01.fa \
--output_dir=output/A1_1 --tmp_dir=tmp/A1_1_tmp --run_mode=both"

thanks again for your patience and kindly reponse!

**Comment by jmtsuji** [comment]
@xushaoyi Yes, you can run multiple HMMs and multiple samples in the same MetAnnotate command. I think that the filepaths should be separated by commas instead of spaces, however.

Just to confirm, have you predicted open reading frames (ORFs) from your raw metagenome reads already, or are you providing nucleotide sequences to MetAnnotate? It looks like you are providing just the nucleotide sequences, because you are running `--do_transeq` -- correct? Are these nucleotide sequences just the raw FastQ files from the sequencer, converted to FastA format? I am just asking to better understand the dataset you are working with.

**Comment by xushaoyi** [comment]
> @xushaoyi Yes, you can run multiple HMMs and multiple samples in the same MetAnnotate command. I think that the filepaths should be separated by commas instead of spaces, however.
> 
> Just to confirm, have you predicted open reading frames (ORFs) from your raw metagenome reads already, or are you providing nucleotide sequences to MetAnnotate? It looks like you are providing just the nucleotide sequences, because you are running `--do_transeq` -- correct? Are these nucleotide sequences just the raw FastQ files from the sequencer, converted to FastA format? I am just asking to better understand the dataset you are working with.

thanks！ yes, I just use the nucleotide sequences after QC and converted to fasta format, should I predict ORFs firstly? and then do the metannotate analysis?

**Comment by jmtsuji** [comment]
@xushaoyi Good to know. It should be fine to use those QC-processed nucleotide sequences directly as input into MetAnnotate. If you plan on doing multiple MetAnnotate runs, you might save time by first predicting ORFs so that MetAnnotate does not have to re-predict the ORFs each time it runs.

**Comment by xushaoyi** [comment]
> @xushaoyi Good to know. It should be fine to use those QC-processed nucleotide sequences directly as input into MetAnnotate. If you plan on doing multiple MetAnnotate runs, you might save time by first predicting ORFs so that MetAnnotate does not have to re-predict the ORFs each time it runs.

good suggestion, I will try, thanks very much! In fact, I followed your suggestions and began to install R packages "metannoviz" on windows, and I checked the code, find something confused. what's the meaning of metannotate_table_filename? is it the original data? or an list of file I input(e.g., A1,A2). looking forward to your reply!
# Load your data
metannotate_data <- read_metannotate_data(metannotate_table_filename)

**Comment by xushaoyi** [comment]
> @xushaoyi Good to know. It should be fine to use those QC-processed nucleotide sequences directly as input into MetAnnotate. If you plan on doing multiple MetAnnotate runs, you might save time by first predicting ORFs so that MetAnnotate does not have to re-predict the ORFs each time it runs.

also I finished one sample, and the result was as follows, is it normal? it seems all of them were unclassified. so sad
![image](https://user-images.githubusercontent.com/40730975/92583139-7c3b6300-f2c4-11ea-9538-0a09ac6181f0.png)

**Comment by jmtsuji** [comment]
@xushaoyi Interesting -- it looks like you got classification results based on the "phylogenetic" method but not from the "sequence" method.

I am wondering if this is due to how you downloaded the RefSeq database. Did you run [`refseq_installation.sh`](https://github.com/MetAnnotate/MetAnnotate/blob/develop/refseq_installation.sh), or did you just manually download the RefSeq database from Zenodo? `refseq_installation.sh` performs a few QC and indexing operations on the database after download before it is ready for use.

See the "Example command line usage" at https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui for an example of using `refseq_installation.sh`.

**Comment by xushaoyi** [comment]
> @xushaoyi Interesting -- it looks like you got classification results based on the "phylogenetic" method but not from the "sequence" method.
> 
> I am wondering if this is due to how you downloaded the RefSeq database. Did you run [`refseq_installation.sh`](https://github.com/MetAnnotate/MetAnnotate/blob/develop/refseq_installation.sh), or did you just manually download the RefSeq database from Zenodo? `refseq_installation.sh` performs a few QC and indexing operations on the database after download before it is ready for use.
> 
> See the "Example command line usage" at https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui for an example of using `refseq_installation.sh`.

infact, I just download refesequence from Zenodo,  ok, I will try refseq_installation.sh, thanks very much!

**Comment by xushaoyi** [comment]
> > @xushaoyi Interesting -- it looks like you got classification results based on the "phylogenetic" method but not from the "sequence" method.
> > I am wondering if this is due to how you downloaded the RefSeq database. Did you run [`refseq_installation.sh`](https://github.com/MetAnnotate/MetAnnotate/blob/develop/refseq_installation.sh), or did you just manually download the RefSeq database from Zenodo? `refseq_installation.sh` performs a few QC and indexing operations on the database after download before it is ready for use.
> > See the "Example command line usage" at https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui for an example of using `refseq_installation.sh`.
> 
> infact, I just download refesequence from Zenodo, ok, I will try refseq_installation.sh, thanks very much!

dear jmtsuji, I followed the suggestion and install refseq database as suggested, with command as follows:
 pbzip2 -d -c metannotate_refseq_db_w_gi_2017_03_01.fa.bz2 > metannotate_refseq_db_w_gi_2017_03_01.fa
   perl ~/metannotate/scripts/cleanDatabase.pl metannotate_refseq_db_w_gi_2017_03_01.fa > list_to_remove.txt
   perl ~/metannotate/scripts/removeFromFasta.pl list_to_remove.txt metannotate_refseq_db_w_gi_2017_03_01.fa > metannotate_refseq_db_w_gi_2017_03_01.fixed.fa
   mv metannotate_refseq_db_w_gi_2017_03_01.fixed.fa metannotate_refseq_db_w_gi_2017_03_01.fa
   sudo ~/.local/bin/esl-sfetch --index metannotate_refseq_db_w_gi_2017_03_01.fa
then, I re-run the command, but still ,the same results (unclassified) were given,is it matters? when I use the result based on the "phylogenetic" method, and do the downstream analysis, error appeared as follows..so sad. maybe can you give me some suggestions? thanks a lot!
![image](https://user-images.githubusercontent.com/40730975/92693914-dbf34600-f378-11ea-8794-6125a118a286.png)

**Comment by xushaoyi** [comment]
> @xushaoyi Interesting -- it looks like you got classification results based on the "phylogenetic" method but not from the "sequence" method.
> 
> I am wondering if this is due to how you downloaded the RefSeq database. Did you run [`refseq_installation.sh`](https://github.com/MetAnnotate/MetAnnotate/blob/develop/refseq_installation.sh), or did you just manually download the RefSeq database from Zenodo? `refseq_installation.sh` performs a few QC and indexing operations on the database after download before it is ready for use.
> 
> See the "Example command line usage" at https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui for an example of using `refseq_installation.sh`.

Dear author, I found that the step of index seems wrong, when I input esl-sfetch, it output command not found. and I found that when I tried this command it worked(add the location for esl-sfetch). so I think I can index them manually, but I don't how to do the following steps by myself. can you told me what should I do next in order to create the tsv file? looking forward to your reply!
![image](https://user-images.githubusercontent.com/40730975/92696600-af412d80-f37c-11ea-84f3-04ffde3d4b5e.png)

**Comment by xushaoyi** [comment]
dear jmtsuji, I followed your suggestion and install refeseq database , with command as follows:
   pbzip2 -d -c metannotate_refseq_db_w_gi_2017_03_01.fa.bz2 > metannotate_refseq_db_w_gi_2017_03_01.fa
   perl ~/metannotate/scripts/cleanDatabase.pl metannotate_refseq_db_w_gi_2017_03_01.fa > list_to_remove.txt
   perl ~/metannotate/scripts/removeFromFasta.pl list_to_remove.txt metannotate_refseq_db_w_gi_2017_03_01.fa > metannotate_refseq_db_w_gi_2017_03_01.fixed.fa
   mv metannotate_refseq_db_w_gi_2017_03_01.fixed.fa Refseq.fa
   sudo ~/.local/bin/esl-sfetch --index Refseq.fa

but the output was still unclassified, it confused me a lot ,and when I used the output to do downstream analysis, error happened, did it due to the absence of  "sequence" method data?
![image](https://user-images.githubusercontent.com/40730975/92711998-a4da6000-f38b-11ea-8bb0-0cdeacb33672.png)

**Comment by xushaoyi** [comment]
> @xushaoyi Interesting -- it looks like you got classification results based on the "phylogenetic" method but not from the "sequence" method.
> 
> I am wondering if this is due to how you downloaded the RefSeq database. Did you run [`refseq_installation.sh`](https://github.com/MetAnnotate/MetAnnotate/blob/develop/refseq_installation.sh), or did you just manually download the RefSeq database from Zenodo? `refseq_installation.sh` performs a few QC and indexing operations on the database after download before it is ready for use.
> 
> See the "Example command line usage" at https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui for an example of using `refseq_installation.sh`.

Dear author, in fact, I  test the example data again, use the test database and refeseq database, the output was normal, it was so confused. so maybe I need to predict ORF as the first step?
python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hmms/RPOB.HMM --reference_database=data/ReferenceTest.fa --output_dir=test_output --tmp_dir=test_tmp --run_mode=both

python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hmms/RPOB.HMM --reference_database=data/Refseq.fa --output_dir=test_output --tmp_dir=test_tmp --run_mode=both

**Comment by xushaoyi** [comment]
> @xushaoyi Interesting -- it looks like you got classification results based on the "phylogenetic" method but not from the "sequence" method.
> 
> I am wondering if this is due to how you downloaded the RefSeq database. Did you run [`refseq_installation.sh`](https://github.com/MetAnnotate/MetAnnotate/blob/develop/refseq_installation.sh), or did you just manually download the RefSeq database from Zenodo? `refseq_installation.sh` performs a few QC and indexing operations on the database after download before it is ready for use.
> 
> See the "Example command line usage" at https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui for an example of using `refseq_installation.sh`.

Dear author, in fact, I  test the example data again, use the test database and refeseq database, the output was normal, it was so confused. so maybe I need to predict ORF as the first step?
python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hmms/RPOB.HMM --reference_database=data/ReferenceTest.fa --output_dir=test_output --tmp_dir=test_tmp --run_mode=both

python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hmms/RPOB.HMM --reference_database=data/Refseq.fa --output_dir=test_output --tmp_dir=test_tmp --run_mode=both

**Comment by xushaoyi** [comment]
> > @xushaoyi Interesting -- it looks like you got classification results based on the "phylogenetic" method but not from the "sequence" method.
> > I am wondering if this is due to how you downloaded the RefSeq database. Did you run [`refseq_installation.sh`](https://github.com/MetAnnotate/MetAnnotate/blob/develop/refseq_installation.sh), or did you just manually download the RefSeq database from Zenodo? `refseq_installation.sh` performs a few QC and indexing operations on the database after download before it is ready for use.
> > See the "Example command line usage" at https://github.com/MetAnnotate/MetAnnotate/tree/develop#recommended-docker-installation-for-command-line-and-soon-web-gui for an example of using `refseq_installation.sh`.
> 
> Dear author, in fact, I test the example data again, use the test database and refeseq database, the output was normal, it was so confused. so maybe I need to predict ORF as the first step?
> python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hmms/RPOB.HMM --reference_database=data/ReferenceTest.fa --output_dir=test_output --tmp_dir=test_tmp --run_mode=both
> 
> python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hmms/RPOB.HMM --reference_database=data/Refseq.fa --output_dir=test_output --tmp_dir=test_tmp --run_mode=both

dear author, after predict ORF by fragGene first, finally it worked and the output was normal, but when I use the metannoviz package, it give the error report like this, so confused,wish your reply!
![image](https://user-images.githubusercontent.com/40730975/92840360-920b6e00-f413-11ea-8a03-5a313d33d264.png)
